AssetInformationListener/BaseFunction.cs

```diff
@@ -0,0 +1,96 @@
+using Amazon.XRay.Recorder.Core;
+using Amazon.XRay.Recorder.Core.Strategies;
+using Amazon.XRay.Recorder.Handlers.AwsSdk;
+using Hackney.Core.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace AssetInformationListener
+{
+    /// <summary>
+    /// Base class used for all functions
+    /// Sets up the necessary DI container for the function.
+    /// Any function-specific registration and configuration should be done in the derived class.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public abstract class BaseFunction
+    {
+        protected readonly static JsonSerializerOptions _jsonOptions = CreateJsonOptions();
+
+        protected IConfigurationRoot Configuration { get; }
+        protected IServiceProvider ServiceProvider { get; }
+        protected ILogger Logger { get; }
+
+        internal BaseFunction()
+        {
+            AWSSDKHandler.RegisterXRayForAllServices();
+
+            var services = new ServiceCollection();
+            var builder = new ConfigurationBuilder();
+
+            Configure(builder);
+            Configuration = builder.Build();
+            services.AddSingleton<IConfiguration>(Configuration);
+
+            services.ConfigureLambdaLogging(Configuration);
+            services.AddLogCallAspect();
+
+            ConfigureServices(services);
+
+            // TODO - Remove if not using DynamoDb
```

Comment:
We can remove this line as we are using Dynamodb

Reply:
It will be when the next ticket is done. This is just setting up the application shell.